clients/csharp-uwp/UWPVoiceAssistantSample/AudioOutput/DialogAudioOutputAdapter.cs

```diff
@@ -4,61 +4,73 @@
+    /// Output audio implementation generic to any DialogAudioOutputStream source that manages the creation and use of
+    /// a Windows AudioGraph for the output of media associated with a DialogResponse, such as text-to-speech output.
     /// </summary>
     public class DialogAudioOutputAdapter
         : IDisposable, IDialogAudioOutputAdapter
     {
+        private static readonly AudioEncodingProperties DefaultEncoding
+            = AudioEncodingProperties.CreatePcm(16000, 1, 16);
+
         private readonly SemaphoreSlim graphSemaphore;
         private readonly AutoResetEvent outputEndedEvent;
-        private readonly object audioOutputStreamsLock;
-        private readonly Queue<DialogAudioOutputStream> audioOutputStreams;
-
+        private readonly ILogProvider log = LogRouter.GetClassLogger();
+        private DialogAudioOutputStream activeOutputStream;
         private AudioGraph graph;
         private AudioFrameInputNode frameInputNode;
+        private AudioEncodingProperties frameInputEncoding;
         private AudioDeviceOutputNode deviceOutputNode;
-        private DeviceWatcher audioOutputDeviceWatcher;
-        private bool firstDeviceEnumerationComplete = false;
-        private bool alreadyDisposed = false;
-        private int activeUtterances = 0;
+        private string lastOutputDeviceId;
+        private bool alreadyDisposed = false;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="DialogAudioOutputAdapter"/> class.
         /// </summary>
         private DialogAudioOutputAdapter()
         {
-            this.audioOutputStreams = new Queue<DialogAudioOutputStream>();
-            this.audioOutputStreamsLock = new object();
             this.outputEndedEvent = new AutoResetEvent(false);
             this.graphSemaphore = new SemaphoreSlim(1, 1);
-            this.InitializeAudioDeviceWatching();
+            this.frameInputEncoding = DefaultEncoding;
+            this.ConfigureOutputDeviceWatching();
         }
 
         /// <summary>
-        /// Raised when all enqueued audio has completed playback.
+        /// Raised when the current active audio playback has finished.
         /// </summary>
         public event Action OutputEnded;
 
         /// <summary>
         /// Gets a value indicating whether the adapter is currently playing audio.
         /// </summary>
-        public bool IsPlaying { get; private set; }
+        public bool IsPlaying { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the encoding to be used for output. Should match the expected format of data being provided.
+        /// </summary>
+        public AudioEncodingProperties OutputEncoding
+        {
+            get => this.frameInputEncoding;
+            set
+            {
+                this.frameInputEncoding = value;
+                _ = this.RegenerateAudioGraphAsync();
```

Comment:
Is it safe to have a large async function that isn't awaited as part of the setter? After reading https://stackoverflow.com/questions/6602244/how-to-call-an-async-method-from-a-getter-or- I wonder if it would be better to change this from a property to a method so we can be sure the audiograph has fully regenerated after setting it?